### PR TITLE
fix: improve error handling for export downloads

### LINKED ISSUE #1357: Feature detection for export downloads is too broad
The try/catch surround the feature detection for export downloads masks other errors, e.g. server errors:

This is the feature detection code:

```
    const isFileSaverSupported = !!new Blob();
```

The try/catch should wrap only this line of code. Other errors should be handled separately.
 
Additionally:

* We should not use an alert to display the error message
* The error message should be localized

### PR DESCRIPTION
**Pull request by testower**
- Separate feature detection try/catch from server error handling
- Replace alert with proper error notifications
- Add localized error messages for browser support and download errors

Closes #1357 

Pending translations for Swedish and Finnish

### CODE REVIEW

File: src/i18n/translations/en.ts [hunk 1]
```diff
@@ -650,4 +650,10 @@ export const messages: MessagesKey = {
   mapLoadingStopsDataText: 'Loading stops...',
   locateStopPoint: 'Locate',
   locateStopPointTooltip: 'Locate on map',
+  exportsDownloadUnsupportedBrowserTitle: 'Browser Not Supported',
```
Comment (testower) [inline src/i18n/translations/en.ts:653]: Swedish: Webbläsaren stöds inte

File: src/i18n/translations/en.ts [hunk 2]
```diff
@@ -650,4 +650,10 @@ export const messages: MessagesKey = {
...
+  exportsDownloadUnsupportedBrowserMessage:
+    'Sorry, your browser does not support file downloads.',
+  exportsDownloadErrorTitle: 'Download Failed',
```
Comment (testower) [inline src/i18n/translations/en.ts:656]: Swedish: Nedladdning misslyckades

File: src/i18n/translations/en.ts [hunk 3]
```diff
@@ -650,4 +650,10 @@ export const messages: MessagesKey = {
...
+  exportsDownloadErrorMessage:
+    'An error occurred while downloading the export. Please try again.',
```
Comment (testower) [inline src/i18n/translations/en.ts:658]: Swedish: Ett fel har uppstått medan filen laddades ner. Försök igen.

File: src/i18n/translations/en.ts (hunk 1 above)
Comment (esuomi) [reply src/i18n/translations/en.ts:653]: Finnish: _Tätä selainta ei tueta_

Literally "this browser is not supported". You can also drop "_Tätä_", but it feels more natural to include it as it brings context to the translation.

File: src/i18n/translations/en.ts [hunk 4]
```diff
@@ -650,4 +650,10 @@ export const messages: MessagesKey = {
   mapLoadingStopsDataText: 'Loading stops...',
   locateStopPoint: 'Locate',
   locateStopPointTooltip: 'Locate on map',
+  exportsDownloadUnsupportedBrowserTitle: 'Browser Not Supported',
+  exportsDownloadUnsupportedBrowserMessage:
+    'Sorry, your browser does not support file downloads.',
```
[the remark this answers is not in the document]
Comment (esuomi) [reply src/i18n/translations/en.ts:655]: Finnish: _Käyttämäsi selain ei tue tiedostojen lataamista._

Literally "The browser you are using does not support downloading of files."

File: src/i18n/translations/en.ts (hunk 2 above)
Comment (esuomi) [reply src/i18n/translations/en.ts:656]: Finnish: _Lataus epäonnistui_

Literally the same :)

File: src/i18n/translations/en.ts (hunk 3 above)
Comment (esuomi) [reply src/i18n/translations/en.ts:658]: Finnish: _Latauksen aikana tapahtui odottamaton virhe. Yritä uudelleen._

Literally "During download an unexpected error occurred. Try again." Finnish isn't naturally polite, we could start the sentence with _Valitettavasti latauksen..._ ("Sadly", "Unfortunately", "Regrettably", "Lamentably") but that's maybe a bit too soft for this. Plus if the sentence starts softly, then the ending should also reflect that with _Yritäthän_ ("Please try...") or _Yrittäisitkö_ ("Could you try...") but that's starts to sound kinda condescending and makes this translation a bit long.

Agglutinative languages, yay? :)

**Pull request merged**